deeptime/markov/msm/tram/_tram.py

```diff
@@ -0,0 +1,708 @@
+import copy
+import warnings
+from typing import Optional
+
+from deeptime.markov.msm import MarkovStateModelCollection
+from deeptime.markov import TransitionCountEstimator, TransitionCountModel
+from deeptime.markov._base import _MSMBaseEstimator
+from deeptime.util import types
+# from _tram_bindings import tram
+from deeptime.markov._tram_bindings import tram
+
+import numpy as np
+import scipy as sp
+
+
+def to_zero_padded_array(arrays, desired_shape):
+    """Pad a list of numpy arrays with zeros to desired shape. Desired shape should be at least the size of the
+    largest np array in the list.
+    Example: arrays=[np.array([1,2]), np.array([3,4,5]), np.array([6])], desired_shape=(4)
+    Returns: np.array([[1, 2, 0, 0], [3, 4, 5, 0], [6, 0, 0, 0]])
+
+    Parameters
+    ----------
+    arrays: array-like(np.array)
+        The list/array of numpy arrays of different shapes. All arrays should have the same number of dimensions.
+    desired_shape: tuple
+        The shape each array should be passed to, having the same number of dimensions as the arrays.
+
+    Returns
+    -------
+    arrays: np.array
+        The passed arrays as one numpy array.
+    """
+    new_array = np.zeros((len(arrays), desired_shape))
+    for i, array in enumerate(arrays):
+        new_array[i, :len(array)] = array
+    return np.ascontiguousarray(new_array)
+
+
+class TRAM(_MSMBaseEstimator):
+    r"""
```

Comment:
docs should be moved from `__init__` to here, do we have a citation for TRAM? In that case you can add it to `references.bib` and cite here

Reply:
some parameters seem to be missing, eg `progress_bar` (how is this handled now?)